Source/Scene/Vector3DTileContent.js

```diff
@@ -0,0 +1,324 @@
+            }).otherwise(function(error) {
+                that.state = Cesium3DTileContentState.FAILED;
+                that.readyPromise.reject(error);
+            });
+        }
+    };
+
+    function createColorChangedCallback(content) {
+        return function(batchId, color) {
+            if (defined(content._primitive)) {
+                content._primitive.updateCommands(batchId, color);
+            }
+        };
+    }
+
+    var sizeOfUint16 = Uint16Array.BYTES_PER_ELEMENT;
+    var sizeOfUint32 = Uint32Array.BYTES_PER_ELEMENT;
+
+    /**
+     * Part of the {@link Cesium3DTileContent} interface.
+     */
+    Vector3DTileContent.prototype.initialize = function(arrayBuffer, byteOffset) {
+        byteOffset = defaultValue(byteOffset, 0);
+
+        var uint8Array = new Uint8Array(arrayBuffer);
+        var magic = getMagic(uint8Array, byteOffset);
+        if (magic !== 'vctr') {
+            throw new DeveloperError('Invalid Vector tile.  Expected magic=vctr.  Read magic=' + magic);
+        }
+
+        var view = new DataView(arrayBuffer);
+        byteOffset += sizeOfUint32;  // Skip magic number
+
+        //>>includeStart('debug', pragmas.debug);
+        var version = view.getUint32(byteOffset, true);
+        if (version !== 1) {
+            throw new DeveloperError('Only Vector tile version 1 is supported.  Version ' + version + ' is not.');
+        }
+        //>>includeEnd('debug');
+        byteOffset += sizeOfUint32;
+
+        var byteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        if (byteLength === 0) {
+            this.state = Cesium3DTileContentState.PROCESSING;
+            this.contentReadyToProcessPromise.resolve(this);
+            this.state = Cesium3DTileContentState.READY;
+            this.readyPromise.resolve(this);
+            return;
+        }
+
+        var featureTableJSONByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var featureTableBinaryByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var batchTableJSONByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var batchTableBinaryByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var indicesByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var positionByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var featureTableString = getStringFromTypedArray(uint8Array, byteOffset, featureTableJSONByteLength);
+        var featureTableJSON = JSON.parse(featureTableString);
+        byteOffset += featureTableJSONByteLength;
+
+        var featureTableBinary = new Uint8Array(arrayBuffer, byteOffset, featureTableBinaryByteLength);
+        byteOffset += featureTableBinaryByteLength;
+
+        var numberOfPolygons = featureTableJSON.NUMBER_OF_POLYGONS;
```

Comment:
There are a bunch of queries to the feature table JSON that should probably be error-checking if these properties actually exist, else throw developer errors.